common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -235,10 +265,9 @@ def create(self, request, uuid):
             if response:
                 return response
         elif entitlement.enrollment_course_run.course_id != course_run_id:
-            self._unenroll_entitlement(
-                entitlement=entitlement,
+            _unenroll_entitlement(
+                course_entitlement=entitlement,
                 course_run_key=entitlement.enrollment_course_run.course_id,
-                user=request.user
             )
             response = self._enroll_entitlement(
```

Comment:
This should probably be wrapped in a transaction with the call to `_unenroll_entitlement`.

Reply:
Also, as I mentioned above with `_unenroll_entitlement`, the `_enroll_entitlement` method itself should probably also be atomic.

Reply:
I saw that you added `@transaction.atomic` to `_enroll_entitlement`, which is great for when that method is called individually. However, here it is being called together with `_unenroll_entitlement`. Would it make sense for these two calls to be made within a `with transaction` block, so that we can roll back to the original state if one of them fails? If not, please add a comment explaining why.